### DISCUSSION TITLE
Will there be further API changes before 1.0?

### DISCUSSION THREAD

**Discussion by Etherian** [nullpub/hkts#17] [Q&A] [answered]
Will there be any major API changes like #9 between now and 1.0? I am considering switching my toy Deno project from fp-ts to hkts and would like to know if I can get started or if I should hold off.

**Comment by baetheus** [comment, marked as answer]
Everything that has tests written is pretty much set for 1.0.0. This includes all files in the root of this repo (basically all the standard algebraic data types and the helper types). I'm mostly done with the optics tests so those apis are solid. The one part of the project I haven't gone over with a fine tooth comb is schematics but it's also the most recently rewritten. There are a few typeclasses that I haven't implemented that likely need some attention. Specifically: Category, Comonad, Profunctor, and ChainRec. I don't have a lot of experience with those yet and there isn't anything in hkts that implements them right now. I don't think the types there will change much but until I implement some stuff I won't know.

tldr: No major changes to the api from its current state are planned. The following things should be noted:
1. At 1.0 the hkts project will be renamed to fun. I've snagged the deno.land/x name for that.
2. I plan to have all the tests done within the week.
3. There are a few type class types that I haven't used enough to say that they are in their final forms.
4. Documentation is going to take awhile (likely a couple of weeks to a month) since I want an example for every export (except for the module instance exports).

**Reply by Etherian** [reply]
Thanks! It sounds like I can go ahead and start converting my project since it mostly just uses basic ADTs and functions. Though I will need Comonad eventually. I would offer to help with Category, Profunctor, and Comonad, but I'm still learning the finer details of Typescript myself.
